### PR TITLE
CC-331: Config option doc updates

### PR DESCRIPTION
**Pull request by shikhar**
Needed a local Kafka build with https://github.com/apache/kafka/pull/1964 to generate the docs using `ConfigDef.toEnrichedRst()`

### CODE REVIEW

**Comment by shikhar** [comment]
@confluentinc/connect for review

File: docs/sink_config_options.rst [hunk 1]
```diff
@@ -10,36 +10,47 @@ Connection
...
 
-``auto.create``
-  Whether to automatically create the destination table based on record schema if it is found to be missing by issuing ``CREATE``.
+``insert.mode``
+  The insertion mode to use. Supported modes are:
 
-  * Type: boolean
-  * Default: false
-  * Importance: medium
+  `insert`
```
Comment (ewencp) [inline docs/sink_config_options.rst]: I assume this is from the upstream doc generation changes, but are we expecting these options to be surrounded in single back ticks? The formatting seems slightly odd compared to the config name.

File: docs/source_config_options.rst [hunk 2]
```diff
@@ -1,34 +1,41 @@
...
+Database
+^^^^^^^^
+
 ``connection.url``
   JDBC connection URL for the database to load.
 
   * Type: string
-  * Default: ""
   * Importance: high
+  * Dependents: ``table.whitelist``,``table.blacklist``
```
Comment (ewencp) [inline docs/source_config_options.rst]: Some of this formatting doesn't seem to work properly, see screenshot attached.
screen shot 2016-10-10 at 3 03 37 pm

Comment (shikhar) [inline docs/source_config_options.rst]: Ack, will fix in the KAFKA-4010 patch & re-generate.

File: docs/sink_config_options.rst (hunk 1 above)
Comment (shikhar) [inline docs/sink_config_options.rst]: This is not from upstream changes, but the doc for this config. Will fix, thanks!

**Comment by shikhar** [comment]
@ewencp updated

**Comment by ewencp** [comment]
LGTM and I'll follow up on the Kafka patch.

**Comment by shikhar** [comment]
sweet ty @ewencp

**Pull request merged**